src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQManagedConnection.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using RabbitMQ.Client;
+
+namespace Enable.Extensions.Queuing.RabbitMQ.Internal
+{
+    internal class RabbitMQManagedConnection
+    {
+        public RabbitMQManagedConnection(IConnection connection)
+        {
+            Connection = connection;
+        }
+
+        public IConnection Connection { get; }
+
+        public int ReferenceCount { get;  private set; }
+
+        public bool Disposed { get; private set; }
+
+        public void DecrementReferenceCount()
+        {
+            ReferenceCount--;
+        }
+
+        public void IncrementReferenceCount()
+        {
+            ReferenceCount++;
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected void Dispose(bool disposing)
+        {
+            if (!Disposed)
+            {
+                if (disposing)
+                {
+                    Connection.Close();
```

Comment:
Do we need this line? Isn't the connection closed when it is disposed?

Reply:
https://www.rabbitmq.com/dotnet-api-guide.html#disconnecting Rabbit MQ docs state that close must be called.